lib/rucio/common/types.py

```diff
@@ -99,3 +99,43 @@ class InternalScope(InternalType):
     '''
     def __init__(self, scope, vo='def', fromExternal=True):
         super(InternalScope, self).__init__(value=scope, vo=vo, fromExternal=fromExternal)
+
+
+class RSEDomainLANDict(TypedDict):
+    read: int
+    write: int
+    delete: int
+
+
+class RSEDomainWANDict(TypedDict):
+    read: int
+    write: int
+    delete: int
+    third_party_copy_read: int
+    third_party_copy_write: int
+
+
+class RSEDomainsDict(TypedDict):
+    lan: dict[str, RSEDomainLANDict]
+    wan: dict[str, RSEDomainWANDict]
+
+
+class RSEProtocolDict(TypedDict):
+    hostname: str
+    scheme: str
+    port: int
+    prefix: str  # FIXME: typing.NotRequired
+    impl: str
+    domains: dict[str, RSEDomainsDict]
+    extended_attributes: dict[str, str]  # FIXME: typing.NotRequired
```

Comment:
As mentioned in a comment above, I believe it’s `prefix: str` and `extended_attributes: Optional[Union[str, dict[str, Any]]]`, but I haven’t checked thoroughly.

Reply:
I think it's my fault: `get_rse_info` returns them to `rse_settings` parameter, so they are required